admin/release.py

```diff
@@ -67,3 +92,207 @@ def make_rpm_version(flocker_version):
         release.extend(remainder)
 
     return rpm_version(version, '.'.join(release))
+
+
+class NotTagged(Exception):
+    """
+    Raised if publising to production and the published version isn't tagged.
+    """
+
+
+class NotARelease(Exception):
+    """
+    Raised if trying to publish to a version that isn't a release.
+    """
+
+
+class Environments(Names):
+    """
+    The environments that documentation can be published to.
+    """
+    PRODUCTION = NamedConstant()
+    STAGING = NamedConstant()
+
+
+@attributes([
+    'documentation_bucket',
+    'cloudfront_cname',
+    'dev_bucket',
+])
+class DocumentationConfiguration(object):
+    """
+    The configuration for publishing documentation.
+
+    :ivar bytes documentation_bucket: The bucket to publish documentation to.
+    :ivar bytes cloudfront_cname: a CNAME associated to the cloudfront
+        distribution pointing at the documentation bucket.
+    :ivar bytes dev_bucket: The bucket buildbot uploads documentation to.
+    """
+
+DOCUMENTATION_CONFIGURATIONS = {
+    Environments.PRODUCTION:
+        DocumentationConfiguration(
+            documentation_bucket="clusterhq-docs",
+            cloudfront_cname="docs.clusterhq.com",
+            dev_bucket="clusterhq-dev-docs"),
+    Environments.STAGING:
+        DocumentationConfiguration(
+            documentation_bucket="clusterhq-staging-docs",
+            cloudfront_cname="docs.staging.clusterhq.com",
+            dev_bucket="clusterhq-dev-docs"),
+}
+
+
+@do
```

Comment:
I went in search of `do` documentation. There aren't any examples but I found the API docs. Still not sure I understand it...yet.
 * https://effect.readthedocs.org/en/latest/api/effect.do.html?highlight=do#module-effect.do